src/freenas/etc/sysctl.conf

```diff
@@ -6,6 +6,7 @@
 # of blocks after they have been in the queues for X seconds.  It is critical
 # that metadelay < dirdelay < filedelay and no fractions are allowed.
 
+kern.corefile=/var/tmp/%N.core
```

Comment:
Actually no, in 11.1 we raise exceptions for various calls when migration of system dataset is being carried out. So if for some reason the system dataset does not exist, migration will fail. Perhaps I should fix that in this PR as well and then we should be safe with `kern.corefile` being set only in sysctl.conf. Thoughts ?

Reply:
kern.corefile is only set at the very end of the setup, I fail to understand how that is an issue.

Reply:
That is an issue in a case where `kern.corefile` is already set to `/var/db/system/cores/%N.core` and we attempt migration of system dataset to another pool. If that fails, `kern.corefile` will still be set to the cores dataset where as it might not even exist. So the underlying problem is that migration should not fail at any time. Please let me know if I missed something or misunderstood it :)

Reply:
Thats a perfectly acceptable situation is system dataset failed. What matters is that it wont coredump to /etc anymore, original problem solved.

Reply:
Yes the original problem is indeed solved. I will remove the setup changes but I do think that we should allow migration of system dataset and if it fails, setup it with the new pool successfully like we are doing for the first time. If it fails, database still reflects that we migrated where as we didn't. Master is good wrt this though. Do you think I should make the appropriate changes for that one in migrate ? This does change the current behaviour of 11.1 though, so not sure if it's intended ?

Reply:
The sysctl change alone is fine.
Everything else is an edge case and this is a old stable branch, lets keep the changes at minimum unless its a critical issue.

Thanks!

Reply:
Makes sense. I'll update the PR. Thank you William